sled-agent/src/services.rs

```diff
@@ -1002,11 +1042,82 @@ impl ServiceManager {
                     self.configure_dns_client(&running_zone, &service).await?;
                     continue;
                 }
+                ServiceType::Maghemite { mode } => {
+                    info!(self.inner.log, "Setting up Maghemite service");
+
+                    smfh.setprop("config/mode", &mode)?;
+                    smfh.setprop("config/admin_host", "::")?;
+
+                    // XXX: tfportd is started in the switch zone when it boots,
+                    // and it will add links for both the rear and front ports
+                    // inside that zone. However, this may take some time.
+                    // Unfortunately [ddmd currently panics if a link is not
+                    // ready][1]. This code is written assuming that [1] is
+                    // fixed: it will configure maghemite to listen on all rear
+                    // ports.
+                    //
+                    // [1]: https://github.com/oxidecomputer/maghemite/issues/53
+
+                    let maghemite_interfaces: Vec<AddrObject> =
+                        if is_gimlet().map_err(|e| {
+                            Error::Underlay(underlay::Error::SystemDetection(e))
+                        })? {
+                            (1..32)
+                                .map(|i| {
+                                    AddrObject::new(
+                                        &format!("tfport{}", i), // XXX rearN soon!
+                                        "ll",
```

Comment:
Use `IPV6_LINK_LOCAL_NAME` from `illumos-utils/src/addrobj.rs`?

Reply:
Done in 85874f9a